File: HackerRank/codetest/ascend.io/truthy.rb
def voidfunc
end

[
  true,
  false,
  0,
  -1,
  1,
  "",
  [],
  {},
  nil,
  voidfunc(),
  undefined_value,
].each do |value|
  if value
    puts "value #{value} was truthy!"
  else
    puts "value #{value} was falsey!"
  end
end

File: HackerRank/codetest/ascend.io/truthy.c
#include "stdbool.h"
#include "stdio.h"

void voidfunc() {

}

int main(int argc, char const *argv[]) {
  if (-1) puts("-1 was truthy");
  if ("") puts("the empty string was truthy");
  int var[0];
  if (var) puts("an empty array was truthy");
  // puts(undefined_value);
  // if (voidfunc()) puts("void return was true");
  return 0;
}
